File: S360_Chat_Sdk/src/main/java/com/android/s360_chat_sdk/S360ChatWidget.java
package com.android.s360_chat_sdk;

import android.content.Context;
import android.content.Intent;
import android.util.AttributeSet;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.widget.RelativeLayout;

public class S360ChatWidget extends RelativeLayout {

   private WebView webView;

   public S360ChatWidget(Context context) {
      super(context);
      init(context);
   }

   public S360ChatWidget(Context context, AttributeSet attrs) {
      super(context, attrs);
      init(context);
   }

   public S360ChatWidget(Context context, AttributeSet attrs, int defStyleAttr) {
      super(context, attrs, defStyleAttr);
      init(context);
   }

   private void init(Context context) {
      // Initialize the WebView
      webView = new WebView(context);
      // Enable JavaScript in the WebView
      WebSettings webSettings = webView.getSettings();
      webSettings.setJavaScriptEnabled(true);
      RelativeLayout.LayoutParams params = new RelativeLayout.LayoutParams(
              LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT);
      webView.setLayoutParams(params);

      // Add WebView to the layout
      addView(webView);
   }

   public void loadUrl(String url) {
      webView.loadUrl(url);
   }

}

File: S360_Chat_Sdk/src/main/java/com/android/s360_chat_sdk/S360ChatWidget.kt
package com.android.s360_chat_sdk

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.webkit.WebView
import android.webkit.WebViewClient
import android.widget.RelativeLayout

class S360ChatWidget : RelativeLayout {

    private val webView: WebView

    constructor(context: Context) : super(context) {
        webView = WebView(context)
        initializeWebView()
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        webView = WebView(context)
        initializeWebView()
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        webView = WebView(context)
        initializeWebView()
    }

    @SuppressLint("SetJavaScriptEnabled")
    private fun initializeWebView() {
        webView.webViewClient = WebViewClient()
        webView.settings.javaScriptEnabled = true

        // Add the WebView to the RelativeLayout
        val params = LayoutParams(
            LayoutParams.MATCH_PARENT,
            LayoutParams.MATCH_PARENT
        )
        addView(webView, params)
    }

    fun init(url: String) {
        webView.loadUrl(url)
    }
}